sdk/dotnet/Pulumi/Deployment/Deployment.cs

```diff
@@ -49,6 +49,10 @@ public static DeploymentInstance Instance
         internal static IDeploymentInternal InternalInstance
             => Instance.Internal;
 
+        private static readonly bool _disableResourceReferences =
+            Environment.GetEnvironmentVariable("PULUMI_DISABLE_RESOURCE_REFERENCES") == "true" ||
```

Comment:
I wonder if this is robust enough. For example, setting to `"True"` wouldn't disable the feature.